src/gui_common/TabButtons.cs

```diff
@@ -241,6 +241,36 @@ public bool OnPreviousSecondaryTab()
         return true;
     }
 
+    /// <summary>
+    ///   Sets an appropriate custom minimum width in order to fit the real sizes of all the included buttons and both
+    ///   indicators on a single line by finding the width of all the nodes that are being used.
+    /// </summary>
+    public void SetCustomMinimumSize()
+    {
+        // Start by adding together the widths of all the tab buttons
+        float newCustomMinimum = 0.0f;
+        foreach (var button in tabButtons)
+        {
+            newCustomMinimum += button.Size.X;
+        }
+
+        // Find the distance between each button (should get 0 if the constant doesn't exist, no need to check)
+        var separationBetweenButtons = tabButtonsContainer.GetThemeConstant("h_separation");
```

Comment:
As this takes in a StringName, the StringName should be cached as this otherwise causes an extra allocation each time.

See for example in SpecifiedInputkey how this is used:
```
private StringName? marginLeftName;
```

as to how I recommend avoiding the allocation on this line each time it is called. Though I just noticed that code doesn't Dispose it correctly... I'll fix that but in the meantime check `ChromaticFilter` for how the dispose should be done. But I'd still prefer the approach where the min size is initialized in this method the first time it is needed.